src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -309,89 +454,34 @@ internal void Log(RpcEvent msg)
             }
             else
             {
-                _logger.Log(logLevel, new EventId(0, rpcLog.EventId), rpcLog.Message, null, (state, exc) => state);
+                _workerChannelLogger.Log(logLevel, new EventId(0, rpcLog.EventId), rpcLog.Message, null, (state, exc) => state);
             }
         }
 
         internal void HandleWorkerError(Exception exc)
         {
-            _logger.LogError(exc, $"Worker encountered an error.");
-
             try
             {
                 if (!_disposed)
                 {
                     _startSubscription?.Dispose();
+                    _process?.Dispose();
 
                     // unlink function inputs
                     foreach (var link in _inputLinks)
                     {
                         link.Dispose();
                     }
-
+                    _workerChannelLogger.LogError(0, exc, $"Language Worker Process: {_process.StartInfo.FileName} exited.");
```

Comment:
Is the event id meaningful here? If so, can we have a constant for clarity? Otherwise, can we use the overload that doesn't take one?

Also, we should be following logging string formatting here and pass the file name as an argument. The string should just be the message template. This allows logging frameworks to store that structure.

Here's some more information: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.1#log-message-template